modules/OCPP201/OCPP201.cpp

```diff
@@ -12,11 +12,57 @@ const std::string CERTS_DIR = "certs";
 
 namespace fs = std::filesystem;
 
-ocpp::v201::SampledValue get_sampled_value(const ocpp::v201::MeasurandEnum& measurand, const std::string& unit,
+types::evse_manager::StopTransactionReason get_stop_reason(const ocpp::v201::ReasonEnum& stop_reason) {
+    switch (stop_reason) {
+    case ocpp::v201::ReasonEnum::DeAuthorized:
+        return types::evse_manager::StopTransactionReason::DeAuthorized;
+    case ocpp::v201::ReasonEnum::EmergencyStop:
+        return types::evse_manager::StopTransactionReason::EmergencyStop;
+    case ocpp::v201::ReasonEnum::EnergyLimitReached:
+        return types::evse_manager::StopTransactionReason::EnergyLimitReached;
+    case ocpp::v201::ReasonEnum::EVDisconnected:
+        return types::evse_manager::StopTransactionReason::EVDisconnected;
+    case ocpp::v201::ReasonEnum::GroundFault:
+        return types::evse_manager::StopTransactionReason::GroundFault;
+    case ocpp::v201::ReasonEnum::ImmediateReset:
+        return types::evse_manager::StopTransactionReason::HardReset;
+    case ocpp::v201::ReasonEnum::Local:
+        return types::evse_manager::StopTransactionReason::Local;
+    case ocpp::v201::ReasonEnum::LocalOutOfCredit:
+        return types::evse_manager::StopTransactionReason::LocalOutOfCredit;
+    case ocpp::v201::ReasonEnum::MasterPass:
+        return types::evse_manager::StopTransactionReason::MasterPass;
+    case ocpp::v201::ReasonEnum::Other:
+        return types::evse_manager::StopTransactionReason::Other;
+    case ocpp::v201::ReasonEnum::OvercurrentFault:
+        return types::evse_manager::StopTransactionReason::OvercurrentFault;
+    case ocpp::v201::ReasonEnum::PowerLoss:
+        return types::evse_manager::StopTransactionReason::PowerLoss;
+    case ocpp::v201::ReasonEnum::PowerQuality:
+        return types::evse_manager::StopTransactionReason::PowerQuality;
+    case ocpp::v201::ReasonEnum::Reboot:
+        return types::evse_manager::StopTransactionReason::Reboot;
+    case ocpp::v201::ReasonEnum::Remote:
+        return types::evse_manager::StopTransactionReason::Remote;
+    case ocpp::v201::ReasonEnum::SOCLimitReached:
+        return types::evse_manager::StopTransactionReason::SOCLimitReached;
+    case ocpp::v201::ReasonEnum::StoppedByEV:
+        return types::evse_manager::StopTransactionReason::StoppedByEV;
+    case ocpp::v201::ReasonEnum::TimeLimitReached:
+        return types::evse_manager::StopTransactionReason::TimeLimitReached;
+    case ocpp::v201::ReasonEnum::Timeout:
+        return types::evse_manager::StopTransactionReason::Timeout;
+    default:
+        return types::evse_manager::StopTransactionReason::Other;
+    }
+}
+
+ocpp::v201::SampledValue get_sampled_value(const ocpp::v201::ReadingContextEnum& reading_context,
```

Comment:
I can not imagine what a `ReadingContextEnum` is about, and it is usually good practice not to include a types type in its type name, like the `Enum` here. Types should be self-explaining if possible

Reply:
Agree. This is an auto generated ocpp::v201 type.